src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,190 @@
+package duke;
+
+import javax.management.Descriptor;
+import java.util.ArrayList;
+
+public class Parser {
+    private User_Input currentType = null;
+    private Ui userInteract;
+    protected TaskList tasks;
+    public boolean isEnd;
+
+
+    public Parser(Ui userInteract, TaskList tasks) {
+        this.userInteract = userInteract;
+        this.tasks = tasks;
+        this.isEnd = false;
+    }
+
+    enum User_Input {
+        TODO,
+        DEADLINE,
+        EVENT,
+        DONE,
+        DELETE,
+        BYE,
+        LIST,
+        FIND
+    }
```

Comment:
I like the use of enum here, make it easy to refer to the different types of command. 👍

Reply:
thank u